packages/rum-core/src/domain/getSessionReplayUrl.spec.ts

```diff
@@ -0,0 +1,74 @@
+import type { RumConfiguration, RumSession, ViewContext } from '@datadog/browser-rum-core'
+import { getSessionReplayUrl, getDatadogSiteUrl } from './getSessionReplayUrl'
+
+describe('getDatadogSiteUrl', () => {
+  const parameters: Array<[string, string | undefined, string]> = [
+    ['datadoghq.com', undefined, 'app.datadoghq.com'],
+    ['datadoghq.com', 'toto', 'toto.datadoghq.com'],
+    ['datad0g.com', undefined, 'dd.datad0g.com'],
+    ['datad0g.com', 'toto', 'toto.datad0g.com'],
+    ['us3.datadoghq.com', undefined, 'us3.datadoghq.com'],
+    ['us3.datadoghq.com', 'toto', 'toto.us3.datadoghq.com'],
+    ['us5.datadoghq.com', undefined, 'us5.datadoghq.com'],
+    ['us5.datadoghq.com', 'toto', 'toto.us5.datadoghq.com'],
+  ]
+
+  parameters.forEach(([site, subdomain, host]) => {
+    it(`should return ${host} for subdomain "${
+      subdomain ?? 'undefined'
+    }" on "${site}" with query params if view is found`, () => {
+      const link = getDatadogSiteUrl({ site, subdomain } as RumConfiguration)
+
+      expect(link).toBe(`https://${host}`)
+    })
+  })
+})
+
+describe('getSessionReplayUrl', () => {
+  const parameters: Array<
+    [string | undefined, { viewId: string; ts: number } | undefined, string | undefined, string]
+  > = [
+    ['session-id-1', undefined, undefined, 'https://app.datadoghq.com/rum/replay/sessions/session-id-1?'],
+    [undefined, undefined, 'toto', 'https://app.datadoghq.com/rum/replay/sessions/no-session-id?error-type=toto'],
+    [
+      'session-id-2',
+      { viewId: 'view-id-1', ts: 1234 },
+      undefined,
+      'https://app.datadoghq.com/rum/replay/sessions/session-id-2?seed=view-id-1&from=1234',
+    ],
+    [
+      'session-id-3',
+      { viewId: 'view-id-2', ts: 1234 },
+      'titi',
+      'https://app.datadoghq.com/rum/replay/sessions/session-id-3?error-type=titi&seed=view-id-2&from=1234',
+    ],
+  ]
+
+  parameters.forEach(([sessionId, viewData, errorType, expected]) => {
+    it(`should return ${expected} for sessionId "${sessionId ?? 'no-session-id'}" with "${JSON.stringify(
+      viewData ?? null
+    )}" as view context`, () => {
+      const session = sessionId
+        ? ({
+            id: sessionId,
+          } as RumSession)
+        : undefined
+      const viewContext = viewData
+        ? ({
+            id: viewData.viewId,
+            startClocks: {
+              timeStamp: viewData.ts,
+            },
+          } as ViewContext)
+        : undefined
+
+      const link = getSessionReplayUrl({ site: 'datadoghq.com' } as RumConfiguration, {
+        viewContext,
+        session,
+        errorType,
+      })
+
+      expect(link).toBe(expected)
+    })
+  })
+})
```

Comment:
<!-- Suggestions are specific requests to improve the subject under review. It is assumed that we all want to do what's best, so these comments are never dismissed as “mere suggestions”, but are taken seriously.  -->
**💬 suggestion:** ‏rework with:

- use object instead of tuple as spec parameter to increase readability and avoid adding cumbersome type 
- document intent of spec case in the spec parameters instead of building a message from the spec parameters
- simplify implementation by providing correct object in spec parameters instead of building them in the test


```suggestion
describe('getSessionReplayUrl', () => {
  const parameters = [
    [{
      testCase: 'session, no view, no error',
      session: { id: 'session-id-1' } as RumSession,
      viewContext: undefined,
      errorType: undefined,
      expected: 'https://app.datadoghq.com/rum/replay/sessions/session-id-1?'
    }],
    [{
      testCase: 'no session, no view, error',
      session: undefined,
      viewContext: undefined,
      errorType: 'toto',
      expected: 'https://app.datadoghq.com/rum/replay/sessions/no-session-id?error-type=toto'
    }],
    [{
      testCase: 'session, view, no error',
      session: { id: 'session-id-2' } as RumSession,
      viewContext: { id: 'view-id-1', startClocks: relativeToClocks(1234 as RelativeTime) },
      errorType: undefined,
      expected: `https://app.datadoghq.com/rum/replay/sessions/session-id-2?seed=view-id-1&from=${getTimeStamp(1234 as RelativeTime)}`,
    }],
    [{
      testCase: 'session, view, error',
      session: { id: 'session-id-3' } as RumSession,
      viewContext: { id: 'view-id-2', startClocks: relativeToClocks(1234 as RelativeTime) },
      errorType: 'titi',
      expected: `https://app.datadoghq.com/rum/replay/sessions/session-id-3?error-type=titi&seed=view-id-2&from=${getTimeStamp(1234 as RelativeTime)}`
    }],
  ]

  parameters.forEach(([{ testCase, session, viewContext, errorType, expected }]) => {
    it(`should build url when ${testCase}`, () => {
      const link = getSessionReplayUrl({ site: 'datadoghq.com' } as RumConfiguration, {
        viewContext,
        session,
        errorType,
      })
      expect(link).toBe(expected)
    })
  })
})
```